src/widgets/InfoBadge.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StyleSheet, View, TouchableHighlight, Text } from 'react-native';
+import Popover from 'react-native-popover-view';
+import Badge from './Badge';
+import { getBadgeData } from '../utilities/getBadgeData';
+import { SUSSOL_ORANGE } from '../globalStyles';
+
+export class InfoBadge extends React.PureComponent {
+  state = {
+    isPopOverVisible: false,
+    rect: {},
+  };
+
+  showPopover() {
+    // Open popover and determine where to open it on canvas
+    this.touchableContainer.measure((ox, oy, width, height, px, py) => {
+      this.setState({
+        isPopOverVisible: true,
+        rect: { x: px, y: py, width, height },
+      });
+    });
+  }
+
+  closePopover() {
+    this.setState({ isPopOverVisible: false });
```

Comment:
if you use `this.setState({rect: null})`, then you can just use the truthy/falsey value of `rect` instead of another state field